File: C++/two-sum.cpp
#include<bits/stdc++.h>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> id;
        for(int i = 0; i < nums.size(); i++) {
            int make = target - nums[i];
            if(id[make]) {
                return {id[make]-1, i};
            }
            id[nums[i]] = i+1;
        }
        return {-1, -1};
    }
};

File: Python/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen_values = {}
        
        for index, value in enumerate(nums):
            if target - value in seen_values:
                return seen_values[target - value], index
            
            else:
                seen_values[value] = index
                
        return -1
